pkgs/build-support/dotnet/build-dotnet-module/default.nix

```
@@ -121,11 +121,13 @@ let
   # this contains all the nuget packages that are implicitly referenced by the dotnet
   # build system. having them as separate deps allows us to avoid having to regenerate
   # a packages dependencies when the dotnet-sdk version changes
-  sdkDeps = dotnet-sdk.packages;
+  sdkDeps = lib.lists.flatten [ dotnet-sdk.packages ];
 
-  sdkSource = mkNugetSource {
-    name = "dotnet-sdk-${dotnet-sdk.version}-source";
-    deps = [ sdkDeps ];
+  sdkSource = let
+    version = dotnet-sdk.version or lib.concatStringsSep "-" dotnet-sdk.versions;
```

```suggestion
    version = dotnet-sdk.version or (lib.concatStringsSep "-" dotnet-sdk.versions);
```
The parentheses are important: e.g. `nix-build -A ArchiSteamFarm`:

```
error: attempt to call something which is not a function but a string

       at /home/xx/.cache/nixpkgs-review/rev-e7cc9afbba6fb8d0657ed70ecabe309a34d92640/nixpkgs/pkgs/build-support/dotnet/build-dotnet-module/default.nix:127:15:

          126|   sdkSource = let
          127|     version = dotnet-sdk.version or (lib.concatStringsSep "-" dotnet-sdk.versions);
             |               ^
          128|   in mkNugetSource {
```